File: composeApp/src/commonMain/kotlin/com/beekeeper/app/domain/model/Weather.kt
package com.beekeeper.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Weather(
    val temperature: Int,
    val humidity: Int,
    val windSpeed: Int,
    val condition: WeatherCondition,
    val description: String
)

enum class WeatherCondition {
    SUNNY,
    PARTLY_CLOUDY,
    CLOUDY,
    RAINY,
    STORMY
}

File: api/app/schemas/weather.py
from pydantic import BaseModel
from enum import Enum


class WeatherCondition(str, Enum):
    SUNNY = "SUNNY"
    PARTLY_CLOUDY = "PARTLY_CLOUDY"
    CLOUDY = "CLOUDY"
    RAINY = "RAINY"
    STORMY = "STORMY"


class WeatherResponse(BaseModel):
    temperature: int
    humidity: int
    wind_speed: int
    condition: WeatherCondition
    description: str
